### PR TITLE
camerad: simplify debayer and fix clamp

### PR DESCRIPTION
**Pull request by geohot**
Shouldn't change behavior, refactors from #24452

### CODE REVIEW

**Comment by geohot** [comment]
Are there debayer tests such that we can trust to merge this?

**Comment by pd0wm** [comment]
There is https://github.com/commaai/openpilot/blob/master/selfdrive/test/process_replay/test_debayer.py which can run on the device.

Install pyopencl with `PYOPENCL_CL_PRETEND_VERSION=2.0 pip install pyopencl`. Once it's in AGNOS we'll add the test to CI.

**Comment by pd0wm** [comment]
The tests shows lots of small changes (0-3).

File: selfdrive/camerad/cameras/real_debayer.cl [hunk 1]
```diff
@@ -13,30 +13,33 @@ const __constant half3 color_correction_2 = (half3)(-0.25277411, -0.05627105, 1.
...
   // poly approximation for s curve
-  half rk = 9 - 100*mp;
-  if (x > mp) {
-    return (rk * (x-mp) * (1-(gamma_k*mp+gamma_b)) * (1+1/(rk*(1-mp))) / (1+rk*(x-mp))) + gamma_k*mp + gamma_b;
-  } else if (x < mp) {
-    return (rk * (x-mp) * (gamma_k*mp+gamma_b) * (1+1/(rk*mp)) / (1-rk*(x-mp))) + gamma_k*mp + gamma_b;
-  } else {
-    return x;
-  }
+  return (x > mp) ?
```
Comment (pd0wm) [inline selfdrive/camerad/cameras/real_debayer.cl:21]: This is now comparing a half3 to a scalar? Something like `select` sounds more appropriate here.

Comment (pd0wm) [reply selfdrive/camerad/cameras/real_debayer.cl:21]: @jwooning pointed out that the ternary operator with a vector is special in OpenCL and is the same as select: https://www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/html/selectionOperator.html

File: selfdrive/camerad/cameras/real_debayer.cl [hunk 2]
```diff
@@ -13,30 +13,33 @@ const __constant half3 color_correction_2 = (half3)(-0.25277411, -0.05627105, 1.
 // tone mapping params
 const half gamma_k = 0.75;
 const half gamma_b = 0.125;
-const half mp_default = 0.01; // ideally midpoint should be adaptive
+const half mp = 0.01; // ideally midpoint should be adaptive
+const half rk = 9 - 100*mp;
```
[the remark this answers is not in the document]
Comment (geohot) [reply selfdrive/camerad/cameras/real_debayer.cl:17]: Hmm, why? I can't imagine it's this change.

**Comment by geohot** [comment]
You think this is okay to merge?

File: selfdrive/camerad/cameras/real_debayer.cl (hunk 1 above)
Comment (geohot) [reply selfdrive/camerad/cameras/real_debayer.cl:21]: Yea I just copied that from Joost's PR

File: selfdrive/camerad/cameras/real_debayer.cl (hunk 2 above)
[the remark this answers is not in the document]
Comment (pd0wm) [reply selfdrive/camerad/cameras/real_debayer.cl:17]: It was really strange, depending on the code that was below removing the `const` qualifier would change the outcome. The change was really small, so it's probably just rounding errors when the compiler computes the `9 - 100 * 0.01` vs when it's done at runtime.

**Comment by pd0wm** [comment]
Think it's good to merge! We can take care of generating new references for test_debayer.

**Pull request merged**